From 566226b9d7a242ca3cafbc26ea9d70ddb40d4135 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Sat, 14 Dec 2013 16:16:01 +0100
Subject: [PATCH 327/327] BUG/MEDIUM: checks: tracking servers must not inherit
 the MAINT flag

If a server is disabled in configuration and another one tracks it,
this last one must not inherit the MAINT flag otherwise it needs to
be explicitly enabled afterwards. Just remove this to fix the issue.
---
 src/cfgparse.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/cfgparse.c b/src/cfgparse.c
index b02bb01..2160243 100644
--- a/src/cfgparse.c
+++ b/src/cfgparse.c
@@ -7038,8 +7038,6 @@ out_uri_auth_compat:
 
 				/* if the other server is forced disabled, we have to do the same here */
 				if (srv->state & SRV_MAINTAIN) {
-					newsrv->state |= SRV_MAINTAIN;
-					newsrv->check.state |= CHK_ST_PAUSED;
 					newsrv->state &= ~SRV_RUNNING;
 					newsrv->check.health = 0;
 					newsrv->agent.health = 0;
-- 
1.8.1.5

